package.json

```diff
@@ -16,32 +16,22 @@
   "exports": {
     "./package.json": "./package.json",
     ".": {
-      "types": "./dist/index.d.ts",
-      "import": "./dist/index.js"
-    },
-    "./datastore": {
-      "types": "./dist/datastore/index.d.ts",
-      "import": "./dist/datastore/index.js"
-    },
-    "./datastore/pinecone": {
-      "types": "./dist/datastore/pinecone/index.d.ts",
-      "import": "./dist/datastore/pinecone/index.js"
-    },
-    "./prompt": {
-      "types": "./dist/prompt/index.d.ts",
-      "import": "./dist/prompt/index.js"
-    },
-    "./model": {
       "types": "./dist/model/index.d.ts",
       "import": "./dist/model/index.js"
     },
-    "./utils": {
-      "types": "./dist/utils/index.d.ts",
-      "import": "./dist/utils/index.js"
+    "./ai-function": {
```

Comment:
<my 2 cents>
having the root export be for models and not everything doesn't make much sense.

barrel files are better for _libraries_ like this one in terms of DX. this current version exposes the implementation's structure which nobody cares about.

for application code, I agree it's much better to avoid barrel file exports, but not for this type of public library code.

for an npm package like this, it should be up to the user whether they want to import from the top-level (which 95% of them will do) or import from the more optimized subdirectories – which given `sideEffects: false` shouldn't have any actual negligible perf benefits anyhow.
</my 2 cents>

Reply:
i still don't agree with this, but i'll think about it a little more.